effect/test_fold.py

```diff
@@ -0,0 +1,60 @@
+
+import operator
+
+from pytest import raises
+
+from effect import (
+    ComposedDispatcher, Effect, Error,
+    base_dispatcher, sync_perform)
+from effect.fold import FoldError, fold_effect
+from effect.testing import SequenceDispatcher
+
+
+def _disp(dispatcher):
+    return ComposedDispatcher([dispatcher, base_dispatcher])
+
+
+def test_fold_effect():
+    """Behaves like foldM."""
```

Comment:
I think this should be unpacked. Not everybody reading this will be familiar enough with haskell to understand this.